src/components/TodoList.tsx

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[]
+  deleteTodo: (id:number) => void
+  tempTodo: Todo | null
+};
+
+export const TodoList: React.FC <Props> = ({
+  todos,
+  deleteTodo,
+  tempTodo,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <div
+          key={todo.id}
+          data-cy="Todo"
+          className={classNames('todo', { completed: todo.completed })}
+        >
+          <label className="todo__status-label">
+            <input
+              onClick={() => {}}
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={todo.completed}
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {todo.title}
+          </span>
+          <button
+            onClick={() => {
+              deleteTodo(todo.id);
+            }}
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+          >
+            ×
+          </button>
+
+          <div data-cy="TodoLoader" className="modal overlay">
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      ))}
+
+      {!!tempTodo && (
+        <div data-cy="Todo" className="todo">
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={tempTodo.completed}
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {tempTodo.title}
+          </span>
+
+          <button type="button" className="todo__remove" data-cy="TodoDelete">
+            ×
+          </button>
+
+          {/* overlay will cover the todo while it is being updated */}
+          <div data-cy="TodoLoader" className="modal overlay is-active">
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      )}
+    </section>
+  );
+};
```

Comment:
please do updates according to comments from the previous task

https://github.com/mate-academy/react_todo-app-loading-todos/pull/1120